docs/spec.md

```diff
@@ -6312,6 +6330,19 @@ function returns `true`. If `x` is not a tensor, returns `None`.
 tensors and returns `num_results` slices of `x` along the axis `axis`.
 If `x` is not a tensor or `dim(x, axis) % num_results != 0`, returns `None`.
 
+* `reduce_implicit_convert(x: Value, source_type: Type, destination_type: Type)
+-> Value` is defined on tensors and returns the converted value of `x` based on
+the `source_type` and `destination_type` as follows:
+
+```python
+def reduce_implicit_convert(x: Value, source_type: Type, destination_type: Type) -> Value:
```

Comment:
Maybe rename to `implicit_convert_for_reduce`?

It looks a little weird to see two "reduce" function nested: `reduce_implicit_covert(reduce(...))`.

Reply:
+1 on further iterating on the name, although my feedback is coming from a bit of a different perspective. 

In my opinion, "reduce_implicit_convert" is a bit too specific to be included in mostly general purpose functions in Notation. E.g. does the name have to include the "reduce" part? There's nothing reduce-specific in this logic, right? What about "implicit"? Is the way how this function is called material for what it does?

Reply:
I agree that some components of the name like `reduce` and `implicit` are not relevant here. The function is similar to StableHLO convert op with additional support for quantization. I have revised the name as `convert_with_quantized_type`. Please let me know. 

Note that once https://github.com/openxla/stablehlo/issues/1576 is resolved we can replace the name with convert. Added a note about that in the spec itself.